src/Talis/Persona/Client/Login.php

```diff
@@ -79,15 +79,21 @@ public function validateAuth()
             throw new \Exception('Payload not json');
         }
 
-        if (
-            !isset($_SESSION[self::LOGIN_PREFIX . ':loginState'])
-            || !isset($payload['state'])
-            || $payload['state'] !== $_SESSION[self::LOGIN_PREFIX . ':loginState']
-        ) {
-            // Error with state - not authenticated
-            $this->getLogger()->error('Login state does not match');
+        if (!isset($_SESSION[self::LOGIN_PREFIX . ':loginState'])) {
+            $this->getLogger()->error('Login state not found on Session');
+            throw new \Exception('Login state does not match (LS01)');
```

Comment:
This code will be presented to the user - are we ok with that?

Reply:
I can remove it, I've changed the logged messages that should be enough.